File: resume_builder.py
from groq import Groq
import os

client = Groq(api_key="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")

def resume_builder(resume, job_description):
    completion = client.chat.completions.create(
        model="llama3-70b-8192",
        messages=[
            {
                "role": "user",
                "content": f"Build a custom resume for this job posting. Here is the resume: {resume} and here is the job description: {job_description}"
            },
        ],
        temperature=1,
        max_tokens=1024,
        top_p=1,
        stream=False,
        stop=None,
    )
    
    return completion.choices[0].message.content

File: not using/resume_builder.go
package main

import (
	"fmt"
	"os"

	"github.com/jpoz/groq"
)

func resumeBuilder(apiKey, resume, jobDescription string) (string, error) {
	var client *groq.Client

	if apiKey == "" {
		apiKey = os.Getenv("GROQ_API_KEY")
	}
	client = groq.NewClient(groq.WithAPIKey(apiKey))
	messages := []groq.Message{
		{
			Role:    "user",
			Content: fmt.Sprintf("Build a custom resume for this job posting. Here is the resume: %s and here is the job description: %s", resume, jobDescription),
		},
	}

	request := groq.CompletionCreateParams{
		Model:       "llama3-70b-8192",
		Messages:    messages,
		Temperature: 1,
		MaxTokens:   1024,
		TopP:        1,
	}

	response, err := client.CreateChatCompletion(request)
	if err != nil {
		return "", err
	}

	return response.Choices[0].Message.Content, nil
}
